integration-tests/tests/src/tests/counter.ts

```diff
@@ -0,0 +1,994 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2024 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import { expect } from "chai";
+import Realm, { BSON, Counter, ObjectSchema, UpdateMode } from "realm";
+
+import { openRealmBeforeEach } from "../hooks";
+import { expectRealmDictionary, expectRealmList, expectRealmSet } from "../utils/expects";
+
+interface IWithCounter {
+  _id: BSON.ObjectId;
+  counter: Counter;
+}
+
+const WithCounterSchema: ObjectSchema = {
+  name: "WithCounter",
+  primaryKey: "_id",
+  properties: {
+    _id: { type: "objectId", default: () => new BSON.ObjectId() },
+    counter: "counter",
+  },
+};
+
+interface IWithNullableCounter {
+  _id: BSON.ObjectId;
+  nullableCounter?: Counter | null;
+}
+
+const WithNullableCounterSchema: ObjectSchema = {
+  name: "WithNullableCounter",
+  primaryKey: "_id",
+  properties: {
+    _id: { type: "objectId", default: () => new BSON.ObjectId() },
+    nullableCounter: "counter?",
+  },
+};
+
+interface IWithDefaultCounter {
+  counterWithDefault: Counter;
+}
+
+const WithDefaultCounterSchema: ObjectSchema = {
+  name: "WithDefaultCounter",
+  properties: {
+    counterWithDefault: { type: "int", presentation: "counter", default: 0 },
```

Comment:
Would this be more interesting if the `default` was different than `0`?

Reply:
Yes it would! 🙂